File: data/multilang_comparison/program_cpp_vector/src/func36.c
#include "program_cpp_vector.h" 
array_t* func36(array_t_param* vars, const unsigned long PATH0, int loopsFactor) {
   size_t pCounter = vars->size;
   if(PATH0 & 1) {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array70 = func37(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array70->id);
      free(params0.data);
      array70->refC--;
      if(array70->refC == 0) {
         free(array70->data);
         free(array70);
         DEBUG_FREE(array70->id);
      }
   }
   else {
      array_t_param params0;
      params0.size = 0;
      params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
      array_t* array76 = func38(&params0, get_path(), loopsFactor);
      DEBUG_RETURN(array76->id);
      free(params0.data);
      array76->refC--;
      if(array76->refC == 0) {
         free(array76->data);
         free(array76);
         DEBUG_FREE(array76->id);
      }
   }
   array_t_param params0;
   params0.size = 0;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   array_t* array81 = func39(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array81->id);
   free(params0.data);
   return array81;
}

File: data/multilang_comparison/program_nim/src/func36.nim
proc func36*(vars: ptr ArrayTParam, PATH0: uint64, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   if (PATH0 and 1'u64) != 0:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array128: ptr ArrayT = func37(addr params0, getPath(), loopsFactor)
      debugReturn(array128.id)
      dealloc(params0.data)
      dec array128.refC
      if array128.refC == 0:
         dealloc(array128.data)
         dealloc(array128)
         debugFree(array128.id)
   else:
      discard
      var params0: ArrayTParam
      params0.size = 0
      params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
      var array138: ptr ArrayT = func38(addr params0, getPath(), loopsFactor)
      debugReturn(array138.id)
      dealloc(params0.data)
      dec array138.refC
      if array138.refC == 0:
         dealloc(array138.data)
         dealloc(array138)
         debugFree(array138.id)
   var params0: ArrayTParam
   params0.size = 0
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   var array143: ptr ArrayT = func39(addr params0, getPath(), loopsFactor)
   debugReturn(array143.id)
   dealloc(params0.data)
   return array143

